pingpong.c

```diff
@@ -251,7 +259,7 @@ int main(int argc, char **argv)
   }
   printf("\n");
 
-  for (size_t i = 0; i < last_cpu; ++i) {
+  for (size_t i = 0; i < last_cpu && i < nr_tested_cores; ++i) {
```

Comment:
That does not seem correct. nr_tested_cores is about counting the number of cores actually tested.
But you make the test i < nr_tested_cores dependent on the core index (i) and not the number of cores actual tested. Based on the CPU_ISSET() below, seems like some CPUs may be skipped.

Reply:
Added a new variable to count the number of cores.